gpt4all-backend/llmodel.h

```diff
@@ -64,20 +63,16 @@ class LLModel {
     virtual void setThreadCount(int32_t /*n_threads*/) {}
     virtual int32_t threadCount() const { return 1; }
 
-    // FIXME: This is unused??
-    const Implementation& getImplementation() const {
-        return *implementation;
+    const Implementation& implementation() const {
+        return *m_implementation;
     }
 
-    // FIXME: Maybe have an 'ImplementationInfo' class for the GUI here, but the DLHandle stuff should
-    // be hidden in cpp file
-    // FIXME: Avoid usage of 'get' for getters
-    static const std::vector<Implementation>& getImplementationList();
-    static const Implementation *getImplementation(std::ifstream& f, const std::string& buildVariant);
+    static const std::vector<Implementation>& implementationList();
+    static const Implementation *implementation(std::ifstream& f, const std::string& buildVariant);
     static LLModel *construct(const std::string &modelPath, std::string buildVariant = "default");
 
 protected:
-    const Implementation *implementation; // FIXME: This is dangling! You don't initialize it in ctor either
+    const Implementation *m_implementation; // This is initialized via Implementation::construct
```

Comment:
You can remove the comment and initialize it to nullptr in the regular constructor...

Reply:
Don't quite see why, but sure, I can do that.